### DISCUSSION TITLE
devel/electron11 and devel/electron12

### DISCUSSION THREAD

**Discussion by grahamperrin** [helloSystem/hello#224] [General]
Spun off from https://github.com/helloSystem/hello/issues/161#issuecomment-946900935

> … All `electron` packages are entirely missing from `pkg search` …

## [devel/electron11](https://www.freshports.org/devel/electron11/)

Deleted. 

FreshPorts is our friend. See the gravestone icon with **RIP** (rest in peace) near the head of the page. The icon is clickable. 

## [devel/electron12](https://www.freshports.org/devel/electron12/)

<https://www.freshports.org/devel/electron12/#packages> packages are currently available for FreeBSD:14:amd64 and FreeBSD:13:amd64. 

### FreeBSD 12.2⋯

1. https://pkg-status.freebsd.org/builds?type=package&jailname=122amd64 both builds _stopped:done:_ at the time of writing
2. http://beefy6.nyi.freebsd.org/jail.html?mastername=122amd64-default show all entries
3. http://beefy6.nyi.freebsd.org/build.html?mastername=122amd64-default&build=dd3b4f6d0e27 is currently highlighted
4. http://beefy6.nyi.freebsd.org/data/122amd64-default/dd3b4f6d0e27/logs/errors/electron12-12.0.9_2.log

![image](https://user-images.githubusercontent.com/192271/138003822-1d239e83-216b-46a0-b07f-1d9b784fa664.png)

FreshPorts finds just one bug, for the DIST option enabled. 

At a glance, the log shows `DIST=off`. I'll send an e-mail to the maintainer.

**Comment by grahamperrin** [comment]
> ### FreeBSD 12.2⋯

For **quarterly** ([currently used by experimental builds of helloSystem](https://github.com/helloSystem/ISO/commit/32ae6b9299b5313874bb3db9d6dacd92f3d7fb95)), the tail of yesterday's <http://beefy2.nyi.freebsd.org/data/122amd64-quarterly/af8aa82d1a32/logs/errors/electron12-12.0.9_2.log>: 

```text
… 
c++ -MMD -MF obj/third_party/blink/renderer/core/core/layout_ng_button.o.d -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1 -DUSE_X11=1 -DOFFICIAL_BUILD -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNO_UNWIND_TABLES -D_GNU_SOURCE -DCR_CLANG_REVISION=\"llvmorg-12-init-16296-g5e476061-1\" -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DBLINK_CORE_IMPLEMENTATION=1 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 -DVK_USE_PLATFORM_XCB_KHR -DGL_GLEXT_PROTOTYPES -DUSE_GLX -DUSE_EGL -DBLINK_IMPLEMENTATION=1 -DINSIDE_BLINK -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DUCHAR_TYPE=uint16_t -DWEBRTC_ENABLE_AVX2 -DWEBRTC_NON_STATIC_TRACE_EVENT_HANDLERS=0 -DWEBRTC_CHROMIUM_BUILD -DWEBRTC_POSIX -DWEBRTC_LINUX -DWEBRTC_BSD -DABSL_ALLOCATOR_NOTHROW=1 -DWEBRTC_USE_BUILTIN_ISAC_FIX=0 -DWEBRTC_USE_BUILTIN_ISAC_FLOAT=1 -DWEBRTC_USE_X11 -DHAVE_SCTP -DNO_MAIN_THREAD_WRAPPING -DGOOGLE_PROTOBUF_NO_RTTI -DGOOGLE_PROTOBUF_NO_STATIC_INITIALIZER -DHAVE_PTHREAD -DSK_CODEC_DECODES_PNG -DSK_CODEC_DECODES_WEBP -DSK_ENCODE_PNG -DSK_ENCODE_WEBP -DSK_USER_CONFIG_HEADER=\"../../skia/config/SkUserConfig.h\" -DSK_GL -DSK_CODEC_DECODES_JPEG -DSK_ENCODE_JPEG -DSK_USE_LIBGIFCODEC -DSK_VULKAN_HEADER=\"../../skia/config/SkVulkanConfig.h\" -DSK_VULKAN=1 -DSK_SUPPORT_GPU=1 -DSK_GPU_WORKAROUNDS_HEADER=\"gpu/config/gpu_driver_bug_workaround_autogen.h\" -DVK_USE_PLATFORM_XCB_KHR -DANGLE_USE_ABSEIL -DABSL_ALLOCATOR_NOTHROW=1 -DV8_COMPRESS_POINTERS -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_DEPRECATION_WARNINGS -DV8_NO_ARGUMENTS_ADAPTOR -DCPPGC_CAGED_HEAP -DLEVELDB_PLATFORM_CHROMIUM=1 -DLEVELDB_PLATFORM_CHROMIUM=1 -DWTF_USE_WEBAUDIO_PFFFT=1 -DUSE_BLINK_V8_BINDING_NEW_IDL_CALLBACK_FUNCTION -DUSE_BLINK_V8_BINDING_NEW_IDL_INTERFACE -DUSE_SYSTEM_LIBJPEG -DV8_COMPRESS_POINTERS -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_DEPRECATION_WARNINGS -DV8_NO_ARGUMENTS_ADAPTOR -DCPPGC_CAGED_HEAP -I../.. -Igen -I../../third_party/perfetto/include -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -I../../third_party/libyuv/include -I../../third_party/jsoncpp/source/include -Igen/shim_headers/opus_shim -Igen/shim_headers/ffmpeg_shim -Igen/shim_headers/openh264_shim -Igen/shim_headers/libpng_shim -Igen/shim_headers/libwebp_shim -I../../third_party/vulkan-deps/vulkan-headers/src/include -Igen/shim_headers/snappy_shim -I../../third_party/khronos -I../../gpu -Igen/third_party/dawn/src/include -I../../third_party/dawn/src/include -I../../third_party/abseil-cpp -I../../third_party/boringssl/src/include -I../../third_party/protobuf/src -Igen/protoc_out -I../../third_party/ced/src -I../../third_party/icu/source/common -I../../third_party/icu/source/i18n -I../../third_party/webrtc_overrides -I../../third_party/webrtc -Igen/third_party/webrtc -I../../third_party/protobuf/src -Igen/net/third_party/quiche/src -I../../net/third_party/quiche/src -I../../third_party/skia -I../../third_party/libgifcodec -I../../third_party/vulkan/include -I../../third_party/vulkan-deps/vulkan-headers/src/include -I../../third_party/angle/include -I../../third_party/angle/src/common/third_party/base -I../../third_party/abseil-cpp -Igen/angle -I../../third_party/angle/include -I../../v8/include -Igen/v8/include -I../../third_party/mesa_headers -I../../third_party/libwebm/source -I../../third_party/leveldatabase -I../../third_party/leveldatabase/src -I../../third_party/leveldatabase/src/include -I../../third_party/iccjpeg -I../../third_party/ots/include -I../../v8/include -Igen/v8/include -I../../v8/include -I../../third_party/zlib -fno-delete-null-pointer-checks -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -pthread -fcolor-diagnostics -fmerge-all-constants -m64 -march=x86-64 -msse3 -Xclang -fdebug-compilation-dir -Xclang . -no-canonical-prefixes -Wall -Wextra -Wimplicit-fallthrough -Wunreachable-code -Wthread-safety -Wextra-semi -Wno-missing-field-initializers -Wno-unused-parameter -Wno-c++11-narrowing -Wno-unneeded-internal-declaration -Wno-undefined-var-template -Wno-ignored-pragma-optimize -Wno-implicit-int-float-conversion -Wno-final-dtor-non-final-class -Wno-builtin-assume-aligned-alignment -Wno-deprecated-copy -Wno-thread-safety-analysis -Wno-thread-safety-attributes -fno-omit-frame-pointer -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Wexit-time-destructors -Wglobal-constructors -O2 -fno-ident -fdata-sections -ffunction-sections -g0 -Xclang -debug-info-kind=constructor -Wshadow -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -Wconversion -Wno-float-conversion -Wno-sign-conversion -Wno-shorten-64-to-32 -Wno-implicit-float-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-enum-float-conversion -DPROTOBUF_ALLOW_DEPRECATED=1 -Wno-shorten-64-to-32 -Wno-header-guard -I/usr/local/include -I/usr/local/include/nss -I/usr/local/include/nspr -I/usr/local/include/libpng16 -I/usr/local/include/libxml2 -I/usr/local/include -I/usr/local/include/libxml2 -std=c++14 -fno-trigraphs -Wno-trigraphs -fno-exceptions -fno-rtti -fvisibility-inlines-hidden -Wno-deprecated-declarations -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -isystem /usr/local/include  -c ../../third_party/blink/renderer/core/layout/ng/layout_ng_button.cc -o obj/third_party/blink/renderer/core/core/layout_ng_button.o
In file included from ../../third_party/blink/renderer/core/layout/ng/layout_ng_button.cc:5:
In file included from ../../third_party/blink/renderer/core/layout/ng/layout_ng_button.h:8:
In file included from ../../third_party/blink/renderer/core/layout/ng/flex/layout_ng_flexible_box.h:9:
In file included from ../../third_party/blink/renderer/core/layout/layout_block.h:29:
In file included from ../../third_party/blink/renderer/core/layout/layout_box.h:30:
In file included from ../../third_party/blink/renderer/core/frame/local_frame.h:44:
In file included from gen/third_party/blink/public/mojom/frame/frame.mojom-blink.h:48:
In file included from gen/third_party/blink/public/mojom/messaging/transferable_message.mojom-blink.h:35:
In file included from gen/third_party/blink/public/mojom/messaging/cloneable_message.mojom-blink.h:53:
../../third_party/blink/renderer/core/messaging/blink_cloneable_message_mojom_traits.h:10:10: fatal error: 'mojo/public/cpp/bindings/array_traits_wtf_vector.h' file not found
#include "mojo/public/cpp/bindings/array_traits_wtf_vector.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
*** Error code 1

Stop.
make: stopped in /usr/ports/devel/electron12
build of devel/electron12 | electron12-12.0.9_2 ended at Wed Oct 20 14:18:59 UTC 2021
build time: 35:10:01
!!! build failure encountered !!!
```

**Reply by grahamperrin** [reply]
http://beefy2.nyi.freebsd.org/build.html?mastername=122amd64-quarterly&build=7d125de7c651 `electron12-12.0.9_2` is building, ~~fifteen~~ thirty-two …

At the tail of the log after thirty-two hours: 

`[ 92% 35449/38463] c++ -MMD -MF obj/third_party/blink/renderer/core/core/document_xslt.o.d -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1 -DUSE_X11=1 -DOFFICIAL_BUILD -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNO_UNWIND_TABLES -D_GNU_SOURCE -DCR_CLANG_REVISION=\"llvmorg-12-init-16296-g5e476061-1\" -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DBLINK_CORE_IMPLEMENTATION=1 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 -DVK_USE_PLATFORM_XCB_KHR -DGL_GLEXT_PROTOTYPES -DUSE_GLX -DUSE_EGL -DBLINK_IMPLEMENTATION=1 -DINSIDE_BLINK -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DUCHAR_TYPE=uint16_t -DWEBRTC_ENABLE_AVX2 -DWEBRTC_NON_STATIC_TRACE_EVENT_HANDLERS=0 -DWEBRTC_CHROMIUM_BUILD -DWEBRTC_POSIX -DWEBRTC_LINUX -DWEBRTC_BSD -DABSL_ALLOCATOR_NOTHROW=1 -DWEBRTC_USE_BUILTIN_ISAC_FIX=0 -DWEBRTC_USE_BUILTIN_ISAC_FLOAT=1 -DWEBRTC_USE_X11 -DHAVE_SCTP -DNO_MAIN_THREAD_WRAPPING -DGOOGLE_PROTOBUF_NO_RTTI -DGOOGLE_PROTOBUF_NO_STATIC_INITIALIZER -DHAVE_PTHREAD -DSK_CODEC_DECODES_PNG -DSK_CODEC_DECODES_WEBP -DSK_ENCODE_PNG -DSK_ENCODE_WEBP -DSK_USER_CONFIG_HEADER=\"../../skia/config/SkUserConfig.h\" -DSK_GL -DSK_CODEC_DECODES_JPEG -DSK_ENCODE_JPEG -DSK_USE_LIBGIFCODEC -DSK_VULKAN_HEADER=\"../../skia/config/SkVulkanConfig.h\" -DSK_VULKAN=1 -DSK_SUPPORT_GPU=1 -DSK_GPU_WORKAROUNDS_HEADER=\"gpu/config/gpu_driver_bug_workaround_autogen.h\" -DVK_USE_PLATFORM_XCB_KHR -DANGLE_USE_ABSEIL -DABSL_ALLOCATOR_NOTHROW=1 -DV8_COMPRESS_POINTERS -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_DEPRECATION_WARNINGS -DV8_NO_ARGUMENTS_ADAPTOR -DCPPGC_CAGED_HEAP -DLEVELDB_PLATFORM_CHROMIUM=1 -DLEVELDB_PLATFORM_CHROMIUM=1 -DWTF_USE_WEBAUDIO_PFFFT=1 -DUSE_BLINK_V8_BINDING_NEW_IDL_CALLBACK_FUNCTION -DUSE_BLINK_V8_BINDING_NEW_IDL_INTERFACE -DUSE_SYSTEM_LIBJPEG -DV8_COMPRESS_POINTERS -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_DEPRECATION_WARNINGS -DV8_NO_ARGUMENTS_ADAPTOR -DCPPGC_CAGED_HEAP -I../.. -Igen -I../../third_party/perfetto/include -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -I../../third_party/libyuv/include -I../../third_party/jsoncpp/source/include -Igen/shim_headers/opus_shim -Igen/shim_headers/ffmpeg_shim -Igen/shim_headers/openh264_shim -Igen/shim_headers/libpng_shim -Igen/shim_headers/libwebp_shim -I../../third_party/vulkan-deps/vulkan-headers/src/include -Igen/shim_headers/snappy_shim -I../../third_party/khronos -I../../gpu -Igen/third_party/dawn/src/include -I../../third_party/dawn/src/include -I../../third_party/abseil-cpp -I../../third_party/boringssl/src/include -I../../third_party/protobuf/src -Igen/protoc_out -I../../third_party/ced/src -I../../third_party/icu/source/common -I../../third_party/icu/source/i18n -I../../third_party/webrtc_overrides -I../../third_party/webrtc -Igen/third_party/webrtc -I../../third_party/protobuf/src -Igen/net/third_party/quiche/src -I../../net/third_party/quiche/src -I../../third_party/skia -I../../third_party/libgifcodec -I../../third_party/vulkan/include -I../../third_party/vulkan-deps/vulkan-headers/src/include -I../../third_party/angle/include -I../../third_party/angle/src/common/third_party/base -I../../third_party/abseil-cpp -Igen/angle -I../../third_party/angle/include -I../../v8/include -Igen/v8/include -I../../third_party/mesa_headers -I../../third_party/libwebm/source -I../../third_party/leveldatabase -I../../third_party/leveldatabase/src -I../../third_party/leveldatabase/src/include -I../../third_party/iccjpeg -I../../third_party/ots/include -I../../v8/include -Igen/v8/include -I../../v8/include -I../../third_party/zlib -fno-delete-null-pointer-checks -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -pthread -fcolor-diagnostics -fmerge-all-constants -m64 -march=x86-64 -msse3 -Xclang -fdebug-compilation-dir -Xclang . -no-canonical-prefixes -Wall -Wextra -Wimplicit-fallthrough -Wunreachable-code -Wthread-safety -Wextra-semi -Wno-missing-field-initializers -Wno-unused-parameter -Wno-c++11-narrowing -Wno-unneeded-internal-declaration -Wno-undefined-var-template -Wno-ignored-pragma-optimize -Wno-implicit-int-float-conversion -Wno-final-dtor-non-final-class -Wno-builtin-assume-aligned-alignment -Wno-deprecated-copy -Wno-thread-safety-analysis -Wno-thread-safety-attributes -fno-omit-frame-pointer -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Wexit-time-destructors -Wglobal-constructors -O2 -fno-ident -fdata-sections -ffunction-sections -g0 -Xclang -debug-info-kind=constructor -Wshadow -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -Wconversion -Wno-float-conversion -Wno-sign-conversion -Wno-shorten-64-to-32 -Wno-implicit-float-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-enum-float-conversion -DPROTOBUF_ALLOW_DEPRECATED=1 -Wno-shorten-64-to-32 -Wno-header-guard -I/usr/local/include -I/usr/local/include/nss -I/usr/local/include/nspr -I/usr/local/include/libpng16 -I/usr/local/include/libxml2 -I/usr/local/include -I/usr/local/include/libxml2 -std=c++14 -fno-trigraphs -Wno-trigraphs -fno-exceptions -fno-rtti -fvisibility-inlines-hidden -Wno-deprecated-declarations -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -isystem /usr/local/include  -c ../../third_party/blink/renderer/core/xml/document_xslt.cc -o obj/third_party/blink/renderer/core/core/document_xslt.o`

**Reply by probonopd** [reply]
Wow, a machine the FreeBSD folks named "beefy" (probably for a reason) is taking 17 hours(!) to build Electron... and counting. Just wow.

**Reply by grahamperrin** [reply]
> and counting.

We _might_ have an outcome in the next ~~eight~~ twenty hours or so. ~~(Assuming _beefy2_ and _beefy16_ have similar capabilities, and so on.)~~

**Reply by probonopd** [reply]
Note to self: never ever _think of_ compiling Chromium/Electron locally.

**Reply by grahamperrin** [reply]
> … compiling Chromium/Electron locally …

I successfully built and packaged one, or both, at least once. It wasn't a requirement, I probably wanted to test a fix for something ahead of the package appearing in the FreeBSD repository.

**Reply by grahamperrin** [reply]
[poudriere: build/timeout and runaway_process | The FreeBSD Forums](https://forums.freebsd.org/threads/poudriere-build-timeout-and-runaway_process.82563/)

* the most recent error differs from <https://github.com/helloSystem/hello/discussions/224#discussioncomment-1511136> the previously noted error.

**Reply by grahamperrin** [reply]
https://github.com/helloSystem/hello/discussions/224#discussioncomment-1536096

> … can we expect it to show up for FreeBSD 12 sooner or later …

See https://pkg-status.freebsd.org/builds?type=package&jailname=122amd64

I can't guess when the next runs will begin.

**Reply by grahamperrin** [reply]
> FreeBSD 12.2⋯

12.2 reached end of life on 2022-03-31.

<https://www.freebsd.org/security/unsupported/>

**Comment by grahamperrin** [comment]
### FreeBSD 13.0⋯

[[package - 130amd64-quarterly][devel/electron12] Failed for electron12-12.0.9_2 in build](https://lists.freebsd.org/archives/freebsd-pkg-fallout/2021-October/123767.html)

**Reply by grahamperrin** [reply]
It's a time-consuming build. 

![image](https://user-images.githubusercontent.com/192271/138569094-528cc5f7-383b-431b-8c38-b8d20280bff5.png)

**Reply by probonopd** [reply]
Do I read "success" there?

**Reply by probonopd** [reply]
And can we expect it to show up for FreeBSD 12 sooner or later as well? https://www.freshports.org/devel/electron12/

**Reply by grahamperrin** [reply]
> Do I read "success" there?

Yes, in the 130amd64 jail. 

> FreeBSD 12

There's the separate thread, https://github.com/helloSystem/hello/discussions/224#discussioncomment-1511136

**Reply by grahamperrin** [reply]
> ### FreeBSD 13.0⋯

13.0 reached end of life on 2022-08-31.

<https://www.freebsd.org/security/unsupported/>

**Comment by probonopd** [comment]
This seems to be a systmatic issue with the Packages system:
If packages fail to build for whatever reason, the last still-working build is not preserved.
This leads to many missing packages frequently, unlike e.g., with Debian, where I never experienced this.

**Reply by grahamperrin** [reply]
Debian is not FreeBSD. 

> many missing packages 

Here, please focus on devel/electron12

**Reply by probonopd** [reply]
OK. Do you know where to download the last built devel/electron12 (from when it was still building, if it ever was)?

**Reply by probonopd** [reply]
https://github.com/tagattie/FreeBSD-Electron/releases has it!

**Reply by grahamperrin** [reply]
> … the last built devel/electron12 …

`/var/cache/pkg/`

Which version(s) are there?

**Comment by probonopd** [comment]
```
FreeBSD% sudo pkg add https://github.com/tagattie/FreeBSD-Electron/releases/download/v12.0.6/electron12-12.0.6-freebsd12-amd64.txz
Fetching electron12-12.0.6-freebsd12-amd64.txz: 100%   80 MiB   6.4MB/s    00:13    
Installing electron12-12.0.6...
Extracting electron12-12.0.6: 100%
=====
Message from electron12-12.0.6:

--
===>   NOTICE:

This port is deprecated; you may wish to reconsider installing it:

Uses Python 2.7 which is EOLed upstream.

It is scheduled to be removed on or after 2020-12-31.
FreeBSD% electron12 
ld-elf.so.1: /usr/local/bin/electron12: Undefined symbol "hb_subset_input_set_retain_gids"

```

**Reply by grahamperrin** [reply]
> electron12-12.0.6

From <https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=258607#c7>: 

>> … Make sure you have electron12-12.0.9_2 as electron12-12.0.9_1 is not supported with harfbuzz-3.0.0. ABI has changed[1], …

**Reply by probonopd** [reply]
Boot Environments to the rescue:

Using 

![image](https://user-images.githubusercontent.com/2480569/138568148-33aff2c4-5836-40e8-aad8-69e0c4f8f6d6.png)

I was able to mount an earlier BE and copy libharfbuzz* from there.
__Runs again!__

**Reply by grahamperrin** [reply]
Smart. 

Next: await the outcome of the current build ▶ https://github.com/helloSystem/hello/discussions/224#discussioncomment-1525501
